dynamic_stack_decider/src/dynamic_stack_decider/sequence_element.py

```diff
@@ -24,13 +24,14 @@ def perform(self, reevaluate=False):
     def pop_one(self):
         """
         Pop a single element of the sequence
-        :return: True when the pop was successful, False when the end of the sequence has been reached
         """
+        assert not self.in_last_element(), ("It is not possible to pop a single element when"
+                                            "the last element of the sequence is active")
         self.current_action_index += 1
-        if self.current_action_index >= len(self.actions):
-            return False
-        else:
-            return True
+
+    def in_last_element(self):
```

Comment:
should it not be called is_last_element?

Reply:
I think the element is the sequence element. So the sequence element has to be in the last stage, because it is the whole sequence.

Reply:
ah okay, yeah now I get it